src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,87 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+        {
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
+            {
+                // Currently, jit is able to handle only these types of objects:
+                // * primitives
+                // * frozen strings
+                // * null
+                //
+                // To add support for mutable objects see https://github.com/dotnet/runtime/pull/76135#issuecomment-1257263126
```

Comment:
Why not do it as part of this PR? It should be just a few extra lines in a few places (and a new JIT/EE interface method if #76135 is merged before this one).